### PR TITLE
GH-582 & GH-575: Updated the sensors to adhere to guidelines

### PR DESCRIPTION
**Pull request by mattleibow**
- Updated the sensors to adhere to guidelines:
   - enum names
   - compas low pass filter is now a param
   - updated docs
   - public constructors on data and events
   - stating an already started sensor will throw
   - sensor use the same units on all platforms

- Added a unit conversion API

### CODE REVIEW

File: DeviceTests/DeviceTests.Shared/Barometer_Shared.cs [hunk 1]
```diff
@@ -29,7 +29,7 @@ void Barometer_ReadingChanged(object sender, BarometerChangedEventArgs e)
 
             var d = await tcs.Task;
 
-            Assert.True(d.Pressure >= 0);
+            Assert.True(d.PressureInHectopascals >= 0);
```
Comment (mattleibow) [inline DeviceTests/DeviceTests.Shared/Barometer_Shared.cs]: Renamed the property

File: Samples/Samples/ViewModel/AccelerometerViewModel.cs [hunk 2]
```diff
@@ -48,14 +48,8 @@ public bool IsActive
...
-        public List<string> Speeds { get; } =
-           new List<string>
-           {
-                "Fastest",
-                "Game",
-                "Normal",
-                "User Interface"
-           };
+        public string[] Speeds { get; } =
+           Enum.GetNames(typeof(SensorSpeed));
```
Comment (mattleibow) [inline Samples/Samples/ViewModel/AccelerometerViewModel.cs]: Using the fancy Enum.GetNames

File: Tests/UnitConverters_Tests.cs [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+        [InlineData(3, 0.0524)]
+        [InlineData(10, 0.1745)]
+        [InlineData(57.2958, 1)]
+        [InlineData(114.5916, 2)]
+        [InlineData(171.8873, 3)]
+        [InlineData(572.9578, 10)]
+        [InlineData(1000, 17.4533)]
+        [InlineData(57295.7795, 1000)]
+        [InlineData(0, 0)]
+        public void DegreesPerSecondToRadiansPerSecond(double degrees, double radians)
```
Comment (mattleibow) [inline Tests/UnitConverters_Tests.cs]: Tests, baby, tests!

File: Xamarin.Essentials/Accelerometer/Accelerometer.shared.cs [hunk 4]
```diff
@@ -17,10 +17,10 @@ public static void Start(SensorSpeed sensorSpeed)
                 throw new FeatureNotSupportedException();
 
             if (IsMonitoring)
-                return;
+                throw new InvalidOperationException("Accelerometer has already been started.");
```
Comment (mattleibow) [inline Xamarin.Essentials/Accelerometer/Accelerometer.shared.cs]: Throw when the sensor is started.

File: Xamarin.Essentials/Accelerometer/Accelerometer.shared.cs [hunk 5]
```diff
@@ -68,19 +68,19 @@ internal static void OnChanged(AccelerometerChangedEventArgs e)
 
     public class AccelerometerChangedEventArgs : EventArgs
     {
-        internal AccelerometerChangedEventArgs(AccelerometerData reading) => Reading = reading;
+        public AccelerometerChangedEventArgs(AccelerometerData reading) => Reading = reading;
```
Comment (mattleibow) [inline Xamarin.Essentials/Accelerometer/Accelerometer.shared.cs]: events are public

File: Xamarin.Essentials/Accelerometer/Accelerometer.shared.cs [hunk 6]
```diff
@@ -68,19 +68,19 @@ internal static void OnChanged(AccelerometerChangedEventArgs e)
...
 
         public AccelerometerData Reading { get; }
     }
 
     public readonly struct AccelerometerData : IEquatable<AccelerometerData>
     {
-        internal AccelerometerData(double x, double y, double z)
+        public AccelerometerData(double x, double y, double z)
```
Comment (mattleibow) [inline Xamarin.Essentials/Accelerometer/Accelerometer.shared.cs]: data is public

File: Xamarin.Essentials/Types/Location.shared.cs [hunk 7]
```diff
@@ -57,46 +57,32 @@ public Location(Location point)
...
-            return final;
+            switch (units)
+            {
+                case DistanceUnits.Kilometers:
+                    return UnitConverters.CoordinatesToKilometers(latitudeStart, longitudeStart, latitudeEnd, longitudeEnd);
+                case DistanceUnits.Miles:
+                    return UnitConverters.CoordinatesToMiles(latitudeStart, longitudeStart, latitudeEnd, longitudeEnd);
+                default:
+                    throw new ArgumentOutOfRangeException(nameof(units));
+            }
```
Comment (mattleibow) [inline Xamarin.Essentials/Types/Location.shared.cs]: Moved the conversion logic into the unit conversions API so we can be cool!

File: Xamarin.Essentials/Types/SensorSpeed.shared.cs [hunk 8]
```diff
@@ -2,9 +2,9 @@
...
     public enum SensorSpeed
     {
-        Fastest,
-        Game,
-        UI,
-        Normal,
+        Default = 0,
+        UI = 1,
+        Game = 2,
+        Fastest = 3,
```
Comment (mattleibow) [inline Xamarin.Essentials/Types/SensorSpeed.shared.cs]: Re ordered enums and renamed values

File: Xamarin.Essentials/Types/UnitConverters.shared.cs [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+    public static class UnitConverters
+    {
+        const double twoPi = 2.0 * Math.PI;
+        const double totalDegrees = 360.0;
+        const double atmospherePascals = 101325.0;
+        const double degreesToRadians = Math.PI / 180.0;
+        const double milesToKilometers = 1.609344;
+        const double milesToMeters = 1609.344;
+        const double kilometersToMiles = 1.0 / milesToKilometers;
+        const double celsiusToKelvin = 273.15;
```
Comment (mattleibow) [inline Xamarin.Essentials/Types/UnitConverters.shared.cs]: constants, baby, constants!

File: Samples/Samples/ViewModel/AccelerometerViewModel.cs (hunk 2 above)
Comment (jamesmontemagno) [reply Samples/Samples/ViewModel/AccelerometerViewModel.cs]: much better!

**Pull request merged**